From 14a6213aef93a91cbbcb1bf3d6d3b93f495e0ec0 Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Fri, 23 Mar 2018 08:25:05 -0500
Subject: [PATCH 3540/5242] MLK-17703-7: drm: imx: dcss: configure output pipe
 according to what sink supports

commit  cab1c1e717dfc573945eca76485002e350167422 from
https://source.codeaurora.org/external/imx/linux-imx.git

Output pipe tables' configuration was hardcoded. This patch will allow the output
pipe to be configured according to what the sink supports.

Also, since there's no way to pass gamut and nonlinearity settings from userspace,
configure the input pipe as REC2020/REC2084.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/dcss/dcss-plane.c |   11 +++--------
 1 file changed, 3 insertions(+), 8 deletions(-)

diff --git a/drivers/gpu/drm/imx/dcss/dcss-plane.c b/drivers/gpu/drm/imx/dcss/dcss-plane.c
index 9746cfd..42da040 100644
--- a/drivers/gpu/drm/imx/dcss/dcss-plane.c
+++ b/drivers/gpu/drm/imx/dcss/dcss-plane.c
@@ -462,18 +462,13 @@ static void dcss_plane_atomic_update(struct drm_plane *plane,
 			  drm_mode_vrefresh(&crtc_state->mode));
 
 	ipipe_cfg.pixel_format = pixel_format;
-	ipipe_cfg.nl = NL_REC709;
+
+	ipipe_cfg.nl = NL_REC2084;
 	ipipe_cfg.pr = PR_FULL;
-	ipipe_cfg.g = G_REC709;
+	ipipe_cfg.g = G_REC2020;
 
 	dcss_crtc_get_opipe_cfg(state->crtc, &opipe_cfg);
 
-	/* apparently the other settins that are read from connector are not good,
-	 * so hardcode */
-	opipe_cfg.nl = NL_REC709;
-	opipe_cfg.pr = PR_FULL;
-	opipe_cfg.g = G_REC2020;
-
 	dcss_hdr10_setup(dcss_plane->dcss, dcss_plane->ch_num,
 			 &ipipe_cfg, &opipe_cfg);
 
-- 
1.7.9.5

